packages/wallet/api/src/marshal-contexts.js

```diff
@@ -0,0 +1,257 @@
+ * with a fresh slot.
+ */
+export const makeExportContext = () => {
+  const myData = {
+    purse: {
+      /** @type {MapStore<string, Purse>} */
+      bySlot: makeScalarMap(),
+      /** @type {MapStore<Purse, string>} */
+      byVal: makeScalarMap(),
+    },
+    payment: {
+      /** @type {MapStore<string, Payment>} */
+      bySlot: makeScalarMap(),
+      /** @type {MapStore<Payment, string>} */
+      byVal: makeScalarMap(),
+    },
+    // TODO: offer, contact, dapp
+  };
+  const sharedData = {
+    /** @type {MapStore<string, unknown>} */
+    bySlot: makeScalarMap(),
+    /** @type {MapStore<unknown, string>} */
+    byVal: makeScalarMap(),
+  };
+
+  /**
+   * @param {string} slot
+   * @param {string} _iface
+   */
+  const slotToVal = (slot, _iface) => {
+    if (sharedData.bySlot.has(slot)) {
+      return sharedData.bySlot.get(slot);
+    }
+    const kind = Object.keys(myData).find(k => slot.startsWith(k));
+    assert(kind, X`bad slot kind: ${slot}`);
+    const id = slot.slice(kind.length + 1);
+    const val = myData[kind].bySlot.get(id); // or throw
+    return val;
+  };
+
+  let unknownNonce = 0;
+
+  const valToSlot = val => {
+    if (sharedData.byVal.has(val)) {
+      return sharedData.byVal.get(val);
+    }
+    for (const kind of Object.keys(myData)) {
+      if (myData[kind].byVal.has(val)) {
+        const id = myData[kind].byVal.get(val);
+        return `${kind}:${id}`;
+      }
+    }
+    unknownNonce += 1;
+    const slot = `unknown:${unknownNonce}`;
+    sharedData.byVal.init(val, slot);
+    sharedData.bySlot.init(slot, val);
+    return slot;
+  };
+
+  const makeSaver = (kind, tables) => {
+    let nonce = 0;
+    return val => {
+      const slot = `${(nonce += 1)}`;
+      tables.bySlot.init(slot, val);
+      tables.byVal.init(val, slot);
+    };
+  };
+
+  return harden({
+    savePurseActions: makeSaver('purse', myData.purse),
+    savePaymentActions: makeSaver('payment', myData.payment),
+    /**
+     * @param {string} id
+     * @param {Purse} purse
+     */
+    initPurseId: (id, purse) => {
+      myData.purse.bySlot.init(id, purse);
+      myData.purse.byVal.init(purse, id);
+    },
+    purseEntries: myData.purse.bySlot.entries,
+    /**
+     * @param {string} id
+     * @param {unknown} val
+     */
+    initBoardId: (id, val) => {
+      sharedData.bySlot.init(id, val);
+      sharedData.byVal.init(val, id);
+    },
+    /**
+     * @param {string} id
+     * @param {unknown} val
+     */
+    ensureBoardId: (id, val) => {
+      if (sharedData.byVal.has(val)) {
+        assert.equal(sharedData.byVal.get(val), id);
+        return;
+      }
+      sharedData.bySlot.init(id, val);
+      sharedData.byVal.init(val, id);
+    },
+    ...makeMarshal(valToSlot, slotToVal),
+  });
+};
+
+const defaultMakePresence = iface => {
+  const severed = `SEVERED: ${iface.replace(/^Alleged: /, '')}`;
+  const thing = /** @type {any} */ (Far(severed, {}));
+  return thing;
```

Comment:
could this be `unknown`? could be typed on the function

also, `thing` is distracting.
```suggestion
  return Far(severed, {});
```